src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -291,15 +291,37 @@ public async Task RefreshAsync(CancellationToken cancellationToken)
                     {
                         applicationSettings = new Dictionary<string, ConfigurationSetting>(_applicationSettings, StringComparer.OrdinalIgnoreCase);
 
-                        foreach (KeyValueChange change in keyValueChanges.Values.Concat(changedKeyValuesCollection))
+                        foreach (KeyValueChange change in keyValueChanges.Values)
+                        {
+                            if (change.ChangeType == KeyValueChangeType.Deleted)
+                            {
+                                applicationSettings.Remove(change.Key);
+                                _logger?.LogInformation(LoggingConstants.SuccessfulKeyValueDeleted + change.Key + " on endpoint: " + AppConfigurationEndpoint);
+                            }
+                            else if (change.ChangeType == KeyValueChangeType.Modified)
+                            {
+                                applicationSettings[change.Key] = change.Current;
+                                _logger?.LogInformation(LoggingConstants.SuccessfulKeyValueModified + change.Current + " on endpoint: " + AppConfigurationEndpoint);
+                            }
+
+                            // Invalidate the cached Key Vault secret (if any) for this ConfigurationSetting
+                            foreach (IKeyValueAdapter adapter in _options.Adapters)
+                            {
+                                adapter.InvalidateCache(change.Current);
+                            }
+                        }
+
+                        foreach (KeyValueChange change in changedKeyValuesCollection)
                         {
                             if (change.ChangeType == KeyValueChangeType.Deleted)
                             {
                                 applicationSettings.Remove(change.Key);
+                                _logger?.LogInformation(LoggingConstants.SuccessfulFeatureFlagDeleted + change.Key + " on endpoint: " + AppConfigurationEndpoint);

```

Comment:
> change.Key  [](http://example.com/codeflow?start=103&length=12)

Key names are not needed for feature flags. We can combine both cases and just say that feature flags were updated.